File: CategoriesSingleton.m
//
//  CategoriesSingleton.m
//  iFixit
//
//  Created by Stefan Ayala on 7/8/13.
//
//

#import "CategoriesSingleton.h"

@implementation CategoriesSingleton

+(CategoriesSingleton*)sharedInstance{
    static dispatch_once_t pred;
    static CategoriesSingleton *shared = nil;
    
    dispatch_once(&pred, ^{
        shared = [[CategoriesSingleton alloc] init];
    });
    return shared;
}

@end

File: CategoriesSingleton.swift
//
//  CategoriesSingleton.m
//  iFixit
//
//  Created by Stefan Ayala on 7/8/13.
//
//

class CategoriesSingleton : NSObject {

    var masterCategoryList: [String: AnyObject] = [:]
    var masterDisplayTitleList: [String: String] = [:]

    class var sharedInstance: CategoriesSingleton {
        struct Static {
            static let instance = CategoriesSingleton()
        }
        return Static.instance
    }

}
